client/lib/posts/post-list-cache-store.js

```diff
@@ -34,7 +34,11 @@ function getCacheKey( options ) {
 }
 
 function get( query ) {
-	var key = getCacheKey( query );
+	// @TODO disable post-list-cache-store
+	console.warn( 'Disable post-list-cache-store. Development purpose.' );
+	return null;
```

Comment:
I think you're probably right, but I'd like to look more. But you're not just disabling in development here. The console.warn and early return should be done in a check for development environment.